avsomr/avsimport.cpp

```diff
@@ -84,14 +84,15 @@ static Score::FileError doImportMSMR(MasterScore* score
                   return err;
 
             // corrections after import
+           QString titleText = baseName;
             Text* title = score->getText(Tid::TITLE);
             if (title)
-                  title->setPlainText(baseName);
+                  title->setPlainText(titleText);
 
             score->setMetaTag("movementNumber", "");
             score->setMetaTag("movementTitle", "");
             score->setMetaTag("source", QFileInfo(filePath).fileName());
-            score->setMetaTag("workTitle", title->plainText());
+            score->setMetaTag("workTitle", titleText);
```

Comment:
was better before

Reply:
before there was a potential error, if there is no `Text * title` element, then there would be a crash. I decided that it is better to write the title in the meta tag regardless of whether there is a `Text * title` element or not.

Reply:
It there is a `title`, it should get used, if there is none, it should get set (to `basename`) and then used.

Reply:
I'm not talking about the value, but about the object `Text* title` (pointer)

Reply:
So?. If `score->getText(Tid::TITLE)` returns something, the pointer is set

Reply:
And its value should get used rather then getting overwritten

Reply:
>  If score->getText(Tid::TITLE) returns something, the pointer is set

that's it, I don’t know `score->getText(Tid::TITLE)` always returns an object or can return nullptr  (for example, if Audiveris doesn’t give a title)

Reply:
That's what the `if (!title)` is good for

Reply:
renamed Text* title to titleElement, maybe it will be clearer

Reply:
Hmm, I see what you mean with 'score->getText(Tid::TITLE)` returning 0 (it indeed may), but not why setting it to a different value if it does not.

Reply:
changed the code